cfme/tests/infrastructure/test_provisioning.py

```diff
@@ -197,12 +198,20 @@ def test_provision_approval(
 
     # Wait for e-mails to appear
     def verify():
-        return (
-            len(filter(
-                lambda mail:
-                "your virtual machine request has completed vm {}".format(normalize_text(vm_name))
-                in normalize_text(mail["subject"]),
-                smtp_test.get_emails())) == len(vm_names)
-        )
+        all_emails = smtp_test.get_emails()
+        for name in vm_names:
+            if len(filter(lambda mail:
+                          "your request to reconfigure virtual machine {} was approved".format(
+                              normalize_text(name)
+                          ) in normalize_text(mail["body"]),
+                          all_emails)) > 1:
+                return False
+            if len(filter(lambda mail:
+                          "virtual machine {} will be available".format(
+                              normalize_text(name)
+                          ) in normalize_text(mail["body"]),
+                          all_emails)) != 1:
+                return False
+        return True
```

Comment:
@bargulg  for scvmm/vmware it doesn't  send multiple email with the same topic. in case we have request for 2 VMs - we get 4 emails: first two emails are that request wasn't approved, but one is for requester and another is for approver, third - is that request was approved and the last one - telling request is completed